src/ray/core_worker/core_worker.h

```diff
@@ -93,12 +105,11 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
 
   /// Shut down the worker completely.
   ///
+  /// This must be called before expected termination of a worker / driver.
+  ///
   /// \return void.
   void Shutdown();
 
-  /// Block the current thread until the worker is shut down.
-  void WaitForShutdown();
```

Comment:
This is unnecessary and make code hard to read. I combined the logic into `Shutdown` (it also logically makes sense the shutdown api blocks until it is completed). There's no other work between Shutdown & WaitForShutdown if you look at existing code. For example;

Driver: Calling ShutdownDriver, which calls Shutdown -> WaitForShutdown within the same function.
Worker: Shutdown stops the task execution loop -> WaitForShutdown is called after task execution loop is stopped.

Reply:
This is a great change!